module/wishlist/helper.js

```diff
@@ -0,0 +1,185 @@
+			let userExtensionDocument = await userExtensionQueries.userExtensionDocument({ userId })
+
+			if (userExtensionDocument) {
+				// If the document exists, update the wishlist by removing the specified item
+				let updatedDocument = await userExtensionQueries.findAndUpdate(
+					{ userId, wishlist: { $elemMatch: { _id: projectTempleteId } } },
+					{ $pull: { wishlist: { _id: projectTempleteId } } },
+					{ new: true }
+				)
+				if (updatedDocument) {
+					return {
+						success: true,
+						message: CONSTANTS.apiResponses.WISHLIST_REMOVED,
+					}
+				} else {
+					throw {
+						message: CONSTANTS.apiResponses.PROJECTTEMPLATES_NOTFOUND,
+						status: HTTP_STATUS_CODE.bad_request.status,
+					}
+				}
+			} else {
+				throw {
+					message: CONSTANTS.apiResponses.USEREXTENSION_NOTFOUND,
+					status: HTTP_STATUS_CODE.bad_request.status,
+				}
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * list wishlist
+	 * @method
+	 * @name list
+	 * @param {String} language  -languageCode
+	 * @param {String}  userId   -userId
+	 * @param {Number} pageSize  -pageSize
+	 * @param {Number} pageNo    -pageNo
+	 * @returns {Object}          list of wishlist
+	 */
+
+	static async list(language, userId, pageSize = 100, pageNo = 1) {
+		try {
+			//Find user's wishlist projectTempleteIds
+			let userExtensionDocument = await userExtensionQueries.userExtensionDocument({ userId })
+
+			if (userExtensionDocument) {
+				//Get the projectTempleteIds
+				let projectTemplateIDs = userExtensionDocument[0].wishlist.map((item) => new ObjectId(item._id))
+				let aggregateData = []
+				//Match Query
+				let matchQuery = {
+					$match: {
+						_id: { $in: projectTemplateIDs },
+					},
+				}
+				aggregateData.push(matchQuery)
+
+				// projection aggregate for multilingual
+				let titleField = language ? `$translation.${language}.title` : '$title'
+				let descriptionField = language ? `$translation.${language}.description` : '$description'
+
+				aggregateData.push(
+					{
+						$project: {
+							_id: 1,
+							title: { $ifNull: [titleField, '$title'] },
+							description: { $ifNull: [descriptionField, '$description'] },
+							externalId: 1,
+							createdAt: 1,
+							categories: 1,
+							'metaInformation.duration': 1,
+						},
+					},
+					{
+						//Pagination
+						$facet: {
+							totalCount: [{ $count: 'count' }],
+							data: [{ $skip: pageSize * (pageNo - 1) }, { $limit: pageSize }],
+						},
+					},
+					{
+						//Count  and project the response data
+						$project: {
+							data: 1,
+							count: {
+								$arrayElemAt: ['$totalCount.count', 0],
+							},
+						},
+					}
+				)
+
+				let projectTempleteDocumetation = await projectTemplateQueries.getAggregate(aggregateData)
```

Comment:
@praveenKDass rename this variable to projectTemplateDocuments

Reply:
Changed @VISHNUDAS-tunerlabs